rsmtool/rsmpredict.py

```diff
@@ -25,6 +28,165 @@
 from .writer import DataWriter
 
 
+def fast_predict(
+    input_features,
+    modeler,
+    df_feature_info,
+    trim_min=None,
+    trim_max=None,
+    trim_tolerance=0.4998,
+    train_predictions_mean=None,
+    train_predictions_sd=None,
+    h1_mean=None,
+    h1_sd=None,
+    logger=None,
+):
+    """
+    Compute predictions for a single instance against given model.
+
+    The main difference between this function and the ``compute_and_save_predictions()``
+    function is that the former is meant for batch prediction and reads
+    all its inputs from disk and writes its outputs to disk. This function,
+    however, is meant for real-time inference rather than batch. To this end,
+    it operates *entirely* in memory. Note that there is still a bit of overlap
+    between the two computation paths since we want to use the RSMTool API
+    as much as possible.
+
+    This function should only be used when the goal is to generate predictions
+    using RSMTool models in production. The user should read everything from disk
+    in a separate thread/function and pass the inputs to this function.
+
+    Note that this function only computes regular predictions, not expected
+    scores.
+
+    Parameters
+    ----------
+    input_features : dict[str, float]
+        A dictionary containing the features for the instance for which to
+        generate the model predictions. The keys should be names of the
+        features on which the model was trained and the values should
+        be the *raw* feature values.
+    modeler : rsmtool.modeler.Modeler object
+        The RSMTool modeler object using which the predictions
+        are to be generated. This object should be created from the already
+        existing model file in the "output" directory of the previously run
+        RSMTool experiment.
+    df_feature_info : pandas DataFrame
+        A pandas dataframe containing the information regarding the model features.
```

Comment:
Here you have DataFrame all lowercase even though you’re referring to the official object name. In a different one you have ‘Data frame’ as if it’s referring to a general frame of data. Not sure if this is intentional but maybe we should refer to it the same way for all of them to keep it uniform. I guess there’s no official convention so maybe I’m just splitting hairs.

Reply:
Good catch. I'll fix this one but you are right in that they are used interchangeably throughout the codebase.